osu.Game.Rulesets.Catch/Objects/Drawables/DrawablePalpableCatchHitObject.cs

```diff
@@ -31,12 +43,42 @@ protected DrawablePalpableCatchHitObject(CatchHitObject h)
                 Origin = Anchor.Centre,
                 Anchor = Anchor.Centre,
             });
+
+            IndexInBeatmap.Value = h.IndexInBeatmap;
         }
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            ScaleContainer.Scale = new Vector2(HitObject.Scale);
+            XBindable.BindValueChanged(x =>
+            {
+                if (!IsOnPlate) X = x.NewValue;
```

Comment:
* why is this `X` set in here instead of the base `DrawableCatchHitObject`?
* what's with the `IsOnPlate` check? that should surely be an assert - objects on plate should never be having their `X` changed, surely? would probably even go as far as to say that `XBindable` should be disabled for plated objects

Reply:
I think the goal is to eventually have the plated objects be their own thing and not DHOs? Which would mean that this check is temporary while that's still the case.

Reply:
`X` is only set for `Palpable` objects because non-palpable hit objects (as currently implemented) require to have `X` position of the `DHO` to be 0, independent of `X` of its hit object, due to relative positioning.

Reply:
Objects on the plate doesn't change X after initialized, but it sets `X` once in the initialization there, and it overwrites the X value previously set by the catcher code.
And yes this code should be removed when caught objects are implemented differently.

Reply:
Let's overlook this one for now :) I'm sure pooling will clean this up a lot.